client/src/app/site/base/base-filter.service/base-filter-list.service.ts

```diff
@@ -292,20 +292,20 @@ export abstract class BaseFilterListService<V extends BaseViewModel> implements
                     if (viewModels && viewModels.length) {
                         let models: FilterModel[] = viewModels.filter(filterFn ?? ((): boolean => true));
                         if (mapFn) {
-                            models = Object.values(models.map(mapFn).mapToObject(model => ({ [model.id]: model })));
+                            models = Object.values(models.map(mapFn).mapToObject(model => ({ [model?.id]: model })));
```

Comment:
Why the questionmark? There's no way for the list in the `getViewModelListObservable` would contain null or undefined values, is there? There should only be models and the models should always contain an id